vignettes/adrs_pcwg3.Rmd

```diff
@@ -0,0 +1,565 @@
+    )
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs %>%
+    arrange(!!!get_admiral_option("subject_keys"),AVISITN,PARAMN,),
+  display_vars = exprs(USUBJID,PARCAT1,PARAMCD, PARAM,PARAMN,AVISIT,AVISITN,ADT,AVALC,AVAL)
+)
+```
+
+### Create last assesment flag for OVRLRESC for dealing with last recorded assessment `Pdu`
+```{r}
+adrs <- adrs %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(USUBJID),                # Group records by Subject ID
+      order = exprs(ADT),                      # Sort records by Assessment Date
+      new_var = LAST_REC_FL,                   # Name of the derived flag for last record
+      mode = "last"                            # Flag only the last record per group
+    ),
+    filter = PARAMCD == "OVRLRESC"             # Restriction: Derive flag only for OVRLRESC records
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs %>%
+    filter(PARAMCD== "OVRLRESC" & LAST_REC_FL=="Y"),
+  display_vars = exprs(USUBJID,PARAMCD,AVISIT,AVISITN,ADT,AVALC,AVAL,LAST_REC_FL)
+)
+```
+
+### Best Overall Response (BOR) and Confirmed Best Overall Response (CBOR)
+
+BOR represents the Best Overall Responses observed during the study, reflecting valid tumor responses such as Complete Response (CR), Partial Response (PR), Stable Disease (SD), and Progressive Disease (PD). 
+
+CBOR, on the other hand, refers to the Confirmed Best Overall Response, requiring sustained responses like CR and PR to meet confirmation criteria, such as persistence over a predefined confirmation period (e.g., 28 days), based on PCWG3 guidelines.
+
+
+For both BOR and CBOR, *Progressive Disease Unconfirmed (PDu)* transitions to **PD** if confirmed as progression using the 2+2 rule. If **PDu remains the last recorded assessment** without follow-up confirmation, it is classified as **SD** in this vignette. However, as a more conservative approach, you may choose to classify it as **PD**. Please refer to your study documentation and protocol requirements to confirm the preferred approach to handling unresolved PDu.
+
+```{r}
+
+pdu_to_sd_event <- event(
+  description = "Transition PDu to SD if it is the last unresolved assessment",
+  dataset_name = "adrs",
+  condition = PARAMCD == "OVRLRESC" & AVALC == "PDu" & LAST_REC_FL =="Y", # PDu → SD
+  set_values_to = exprs(AVALC = "SD")
+)
+
+pdu_to_pd_event <- event_joined(
+  description = "Transition PDu to PD if progression is confirmed by subsequent PD",
+  dataset_name = "adrs",
+  join_vars = exprs(AVALC, ADT),            
+  join_type = "after",                      # Look at subsequent assessments after current record
+  first_cond_upper = AVALC.join == "PD" & ADT.join > ADT, # Next record confirms PD with later date
+  condition = PARAMCD == "OVRLRESC" & AVALC == "PDu",     # Current record is PDu
+  set_values_to = exprs(AVALC = "PD")                     # Transition PDu → PD
+)
```

Comment:
Not clear to me if this PDu on last assessment is converted to  -> SD is  converted in actual derived OVRLRESPC record avalc? can this be handled directly on the BOR?